File: frontend/src/types/post.ts
export interface Post {
  id: number
  userId: number
  username: string
  displayName: string
  content: string
  createdAt: string
  updatedAt: string
  likeCount: number
  likedByMe: boolean
  commentCount: number
}

export interface PostPage {
  items: Post[]
  nextCursor: string | null
}

export interface LikeStatus {
  likeCount: number
  likedByMe: boolean
}

File: backend/src/main/java/com/example/murmur/post/Post.java
package com.example.murmur.post;

import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Post {

    private Long id;
    private Long userId;
    private String content;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private String username;
    private String displayName;
}
